easy_profiler_core/profile_manager.h

```diff
@@ -278,20 +511,33 @@ class chunk_allocator
     void serialize(profiler::OStream& _outputStream)
     {
         // Chunks are stored in reversed order (stack).
-        // To be able to iterate them in direct order we have to invert chunks list.
+        // To be able to iterate them in direct order we have to invert the chunks list.
         m_chunks.invert();
 
-        // Iterate over chunks and perform blocks serialization
-        auto current = m_chunks.last;
+        // Each chunk is an array of N bytes that can hold between
+        // 1(if the list isn't empty) and however many elements can fit in a chunk,
+        // where an element consists of a payload size + a payload as follows:
+        // elementStart[0..1]: size as a uint16_t
+        // elementStart[2..size-1]: payload.
+        
+        // The maximum chunk offset is N-sizeof(uint16_t) b/c, if we hit that (or go past),
+        // there is either no space left, 1 byte left, or 2 bytes left, all of which are
+        // too small to cary more than a zero-sized element.
+        constexpr int_fast32_t MAX_CHUNK_OFFSET = (int_fast32_t)N-sizeof(uint16_t);
```

Comment:
Unfortunately `constexpr` keyword is not supported by MSVC2013.
Please, consider creating a `static const` member instead of `constexpr` value...

Reply:
I thought the requirements for this library was a C++11 compliant compiler? I can make the change, though.

Reply:
MSVC2013 is C++11 partly compatible, but it supports the most of features needed by easy_profiler

Reply:
@cas4ey Gotchya